tests/python_package_test/test_sklearn.py

```diff
@@ -1337,6 +1338,60 @@ def test_verbosity_is_respected_when_using_custom_objective(capsys):
     assert "[LightGBM] [Warning] Unknown parameter: nonsense" in capsys.readouterr().out
 
 
+def test_fit_only_raises_num_rounds_warning_when_expected(capsys):
+    X, y = make_synthetic_regression()
+    base_kwargs = {
+        "num_leaves": 5,
+        "verbosity": -1,
+    }
+
+    # no warning: no aliases, all defaults
+    reg = lgb.LGBMRegressor(**base_kwargs).fit(X, y)
+    assert reg.n_estimators_ == 100
+    assert_silent(capsys)
+
+    # no warning: no aliases, just n_estimators
+    reg = lgb.LGBMRegressor(**base_kwargs, n_estimators=2).fit(X, y)
+    assert reg.n_estimators_ == 2
+    assert_silent(capsys)
+
+    # no warning: 1 alias + n_estimators (both same value)
+    reg = lgb.LGBMRegressor(**base_kwargs, n_estimators=3, n_iter=3).fit(X, y)
+    assert reg.n_estimators_ == 3
+    assert_silent(capsys)
+
+    # no warning: 1 alias + n_estimators (different values... value from params should win)
+    reg = lgb.LGBMRegressor(**base_kwargs, n_estimators=3, n_iter=4).fit(X, y)
+    assert reg.n_estimators_ == 4
+    assert_silent(capsys)
+
+    # no warning: 2 aliases (both same value)
+    reg = lgb.LGBMRegressor(**base_kwargs, n_iter=3, num_iterations=3).fit(X, y)
+    assert reg.n_estimators_ == 3
+    assert_silent(capsys)
+
+    # no warning: 4 aliases (all same value)
+    reg = lgb.LGBMRegressor(**base_kwargs, n_iter=3, num_trees=3, nrounds=3, max_iter=3).fit(X, y)
+    assert reg.n_estimators_ == 3
+    assert_silent(capsys)
+
+    # warning: 2 aliases (different values... "num_iterations" wins because it's the main param name)
+    # with pytest.warns(UserWarning, match="LightGBM will perform up to 5 boosting rounds"):
+    #     reg = lgb.LGBMRegressor(**base_kwargs, num_iterations=-8, n_iter=6).fit(X, y)
```

Comment:
Please remove these commented lines.

Reply:
😅 ah sorry, left this in from testing. Thanks as always for your attention to detail! Removed in https://github.com/microsoft/LightGBM/pull/6579/commits/e5de60ab0674eeb1ac0e4dc03f2b201d624d2d79